### PR TITLE
Addition of FSync

### PR DESCRIPTION
**Pull request by jeniawhite**
### Explain the changes
1. Added FSync for I/O flows

### Issues: Fixed #xxx / Gap #xxx
1. Multipart upload as temp files that are renamed
2. FSync on reads for dual access when needed
3. Marking of FSynced files in the FS

### Testing Instructions:
1.

### CODE REVIEW

**Review by liranmauda — APPROVED** [review]
Just one question:
Do you want to keep the close of the dirs in the finally?
Why not use the same logic that we use on files and close it on the try block +  make sure with null that dir (but keep the finally just in case we error on the way)?

**Review by BarakSason — COMMENTED** [review]
Personally, I'd appreciate some more info in the commit message (why is this needed? High-level explanation on how is this implemented?).
Some in-code documentation would be appreciated as well.

**Review by guymguym — CHANGES_REQUESTED** [review]
Since our "update directory" code paths never really have the directory open with opendir (beccause it's only useful for reading entries one-by-one), then I think it makes more sense for fsync to be implemented once as an instance method for FileWrap - which you did, and second as a top api function (like rename) that takes a file/dir path and does open+fsync+close (instead of the DirWrap function). This will also simplify the error handling code needed from js. WDYT?

**Comment by jeniawhite** [comment]
@liranmauda Removed the finally scope code due to usage of a function instead of opening a wrapper.
@BarakSason Please elaborate more on the information that is missing, you can read more at https://man7.org/linux/man-pages/man2/fdatasync.2.html to understand the high-level design, why it is used, and what the method does.

**Review by liranmauda — APPROVED** [review]
LGTM

**Review by guymguym — APPROVED** [review]

File: src/sdk/namespace_fs.js [hunk 1]
```diff
@@ -612,6 +612,7 @@ class NamespaceFS {
             // TODO use file xattr to store md5_b64 xattr, etc.
             const stat = await nb_native().fs.stat(fs_account_config, upload_path);
             await nb_native().fs.rename(fs_account_config, upload_path, file_path);
+            await nb_native().fs.fsync(fs_account_config, path.dirname(upload_path));
```
Comment (guymguym) [inline src/sdk/namespace_fs.js:615]: @jeniawhite Wait, isn't this the wrong path? we should be calling fsync on the target dir of the rename, which would be `path.dirname(file_path)` in this scope

Comment (guymguym) [reply src/sdk/namespace_fs.js:615]: There's a link() call few lines above this line, that I think needs fsync too - and there it's indeed on the dirname of upload_path. Generally it seems that we want to follow any update to the filesystem with an fsync call to the directory of the updated file... that includes all these natiive fs calls - rename, link, unlink, mkdir, rmdir, writeFile, open('w'). We also said we want to have a control flag to turn off fsync.

**Pull request merged**